src/hats.ts

```diff
@@ -380,3 +415,58 @@ function removeHat(hat: Hat, event: TransferSingle): void {
   hatBurnedEvent.hat = hatIdToHex(event.params.id);
   hatBurnedEvent.save();
 }
+
+function createDummyTree(treeDomain: string): void {
+  let tree = new Tree(treeDomain);
+  tree.save();
+}
+
+function createDummyHats(hatId: string, contractAddress: Address): void {
+  for (let i = 10; i < hatId.length; i += 4) {
+    let currentHatId = hatId.substring(0, i).padEnd(66, "0");
+    let hat = Hat.load(currentHatId);
+    if (hat === null) {
+      createDummyHat(currentHatId, contractAddress);
+    }
+
+    let domainAtNextLevel = hatId.substring(i, i + 4);
+    if (domainAtNextLevel == "0000") {
+      break;
+    }
+  }
+}
+
+function createDummyHat(hatId: string, contractAddress: Address): void {
+  let hat = new Hat(hatId);
+  hat.prettyId = hatIdHexToPrettyId(hatId);
+  hat.createdAt = null;
+  hat.wearers = [];
+  hat.details = "";
+  hat.maxSupply = BigInt.fromI32(0);
+  hat.eligibility = "0x0000000000000000000000000000000000000000";
+  hat.toggle = "0x0000000000000000000000000000000000000000";
+  hat.mutable = false;
+  hat.imageUri =
```

Comment:
Curious what your thinking is here. I would have thought we'd leave the imageUri null just like details, but it's possible I'm missing something.

Reply:
I thought mirroring what the contract will give you with `viewHat`. But on a second thought, all other "real" hats store what is actually stored in the hat itself, so I think you're right. More consistent to have it `null`